### ISSUE TITLE
Docker for Windows (and Hyper-V) not working after update to Windows 10 1709

### ISSUE THREAD

**Issue by SCWells72** [docker/for-win#1221]
### Expected behavior

Docker starts fine after updating to Windows 10 1709.

### Actual behavior

Docker fails after updating Windows to 1709 with:

```
Unable to create: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Hyper-V encountered an error trying to access an object on computer 'MYCOMPUTERNAME' because the object was not found. The object might have been deleted. Verify that the Virtual Machine Management service on the computer is running.
at New-Switch, <No file>: line 131
at <ScriptBlock>, <No file>: line 385
   at Docker.Backend.ContainerEngine.Linux.DoStart(Settings settings, String daemonOptions) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Backend\ContainerEngine\Linux.cs:line 246
   at Docker.Backend.ContainerEngine.Linux.Restart(Settings settings, String daemonOptions) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Backend\ContainerEngine\Linux.cs:line 182
   at Docker.Core.Pipe.NamedPipeServer.<>c__DisplayClass9_0.<Register>b__0(Object[] parameters) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Core\pipe\NamedPipeServer.cs:line 47
   at Docker.Core.Pipe.NamedPipeServer.RunAction(String action, Object[] parameters) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Core\pipe\NamedPipeServer.cs:line 145
```

### Information

  - Diagnostic ID 71BC63BB-E7FE-4D08-910F-237D3872E478/2017-10-18_14-31-46
  - Windows 10 1709 on Surface Pro 2017

So to be fair, it doesn't appear that Hyper-V made it through the OS update very well, but since I only use Hyper-V to run Docker for Windows, and since Microsoft support has so far been completely useless, I thought I'd check in here to see if a) anyone else had hit this issue; b) anyone might have thoughts on how to get back into a good state.

I've tried disabling and re-enabling Hyper-V completely (of course with the requisite reboots in between), uninstalling and reinstalling Docker for Windows, installing the edge version of Docker for Windows, resetting the Windows network interfaces completely, and even doing a system restore to the previous state, and at this point it's just not working.

As noted above, the core issue appears to be Hyper-V. The Hyper-V Host Compute Service and Hyper-V Virtual Machine Management service are both reported as Running, but even Hyper-V Manager can't connect to Hyper-V on the local machine to manage it.

Anyone else seeing this? If so, any thoughts on how to get back into a good working state? Thanks in advance!

**Comment by craigotis** [comment]
I'm not sure if I'm seeing the same issue as you, but since updating to 1709 last night, I can no longer launch a Docker QuickStart Terminal. I get two UAC prompts, which I accept, but it fails regardless with:
```
Starting "default"...
(default) Check network to re-create if needed...
Unable to start the VM: C:\Program Files\Oracle\VirtualBox\VBoxManage.exe startvm default --type headless failed:
VBoxManage.exe: error: The virtual machine 'default' has terminated unexpectedly during startup with exit code 1 (0x1).  More details may be available in 'C:\Users\Craig\.docker\machine\machines\default\default\Logs\VBoxHardening.log'
VBoxManage.exe: error: Details: code E_FAIL (0x80004005), component MachineWrap, interface IMachine

Details: 12:01:00.825297 NAT: DHCP offered IP address 10.0.2.15
Looks like something went wrong in step ´Checking status on default´... Press any key to continue...
```

**Comment by SCWells72** [comment]
That sounds a bit different, @craigotis. It looks like you're using VirtualBox whereas I'm using using Windows 10 Hyper-V which has seemingly stopped working. It's possible the root cause is the same, of course...some change in how virtualization is allowed to occur in Windows 10 after the update perhaps?

**Comment by jasonbivins** [comment]
@SCWells72 
Try removing Hyper-V and the Containers windows features - then re-enabling through Powershell. I've seen a few people have better luck doing it that way than through the GUI

   ``` Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Hyper-V -All ``` 
	• Reboot, then enable the containers feature.
   ``` Enable-WindowsOptionalFeature -Online -FeatureName Containers -All ```

**Comment by SCWells72** [comment]
Thanks for the suggestion, @jasonbivins, but same result. I have another Surface Pro...exact same model...on which I'd never enabled Hyper-V and which I also updated to 1709 this morning. I decided to enable Hyper-V to see if the same problem occurred, and it didn't. I'm able to access Hyper-V on the local machine just fine there. It appears that something happened when upgrading the active Hyper-V on the other machine and it just can't get back into a good state. I fear I'm going to have to reload the OS entirely to get it back which just seems silly.

Any other thoughts on where to look for error logging during Hyper-V startup? I scanned the Event Viewer and didn't see anything suspicious, but this is definitely not my forte. It's possible I just don't know where to look.

**Comment by compkickers** [comment]
Ive got the exact same issue, but i came accross this when rs3 was an insider build.  Does your hyper v work just after you have installed it?  Mine does but then after a reboot it dies everytime.

Its def a Hyper-V issue though, would really appreciate some ideas?

JK

**Comment by SCWells72** [comment]
@compkickers mine doesn't work at all anymore. Prior to this morning, zero problems. Since the 1709 update, Hyper-V simply doesn't start any longer even though the services are both shown as running.

**Comment by jasonbivins** [comment]
@SCWells72 
According to the logs and your system events - HyperVisor is starting, and we're able to create and delete the DockerNAT switch without error.
Do you see anything useful in the event viewer under Applications and Services > Microsoft > Windows >Hyper-V
Do you get an error when Hyper-V won't start?

**Comment by SCWells72** [comment]
@jasonbivins I'm not sure I understand your most recent comment. I haven't provided any logs. Were you responding to someone else?

And there are no errors from Hyper-V initialization. In fact, if you open the Services control panel it shows both services (Hyper-V Host Compute Service and Hyper-V Virtual Machine Management) as Started. However, if I tried to use Hyper-V Manager to connect to the local Hyper-V instance it says:

"An error occurred while attempting to connect to server 'MyHostName'. Check that the Virtual Machine Management service is running and that you are authorized to connect to the server. Hyper-V encountered an error trying to access an object on computer 'MyHostName' because the object was not found. The object might have been deleted. Verify that the Virtual Machine Management service on the computer is running."

If I try to start the services from a admin CLI, they show as running:

```
> net start vmcompute
The requested service has already been started.

> net start vmms
The requested service has already been started.
```

**Comment by rvprasad** [comment]
I too encountered this issue after updating to Windows 10 build 1709.  Diagnostic ID for the corresponding crash report is 17A5BBF0-D955-4DB7-A538-CEF004DD5BA1/2017-10-19_00-52-09.

**Comment by SCWells72** [comment]
@rvprasad I think we're seeing slightly different issues. In my case Hyper-V never even starts successfully. It looks like you're able to get things to start but not to stay started. Still, there's obviously some general class of issues with this stuff in 1709 that weren't present before. It doesn't fix my issue, but I do prefer that mine isn't a total one-off because it raises the likelihood of a resolution!

**Comment by SCWells72** [comment]
Okay, so here's a WTF update...I went into the office today and it all started just fine on my work network. Doesn't work at all on my home network, though. Anyone have any idea why that might be the case? Again, the main culprit here is Hyper-V. Is there perhaps something in the VM created by Docker that's tied to a specific network on which it's created?

**Comment by jasonbivins** [comment]
@SCWells72     The diagnostic ID you provided gives me Docker logs and a few relevant system events.
You can try clearing out your virtual switches, it sounds like one of them is stuck or was using an outdated driver.  I've also read through some forums that Resetting Networking in Windows 10 helps, but it's more of a nuclear scorched earth option:  https://www.howtogeek.com/265870/how-to-reset-your-entire-network-in-windows-10-and-start-from-scratch/

**Comment by jasonbivins** [comment]
@rvprasad Let's create the MobyLinuxVM manually to see if the issue is directly with Hyper-V, or to see if we can get a more specific error.  Make sure you run Powershell as an Admin 
``` powershell -ExecutionPolicy ByPass -File "C:\Program Files\Docker\Docker\resources\MobyLinux.ps1" -create ```

**Comment by compkickers** [comment]
Yeah does sound like we have a few causes of the same outcome here, ive had Hyper V issues ever since 1709 insider.  My vms always fail to start from the worker service failing.  It seems some people here are able to get Hyper V to run sometimes but not with Moby linux, for me its all Vms in 1709.  I only use hyper v for docker so i tried docker first then other vms.  But i still have the same error msgs in my Docker client, but as some have pointed out it all points back to Hyper-V.  Ive also posted on MS forums in the hopes they can help, im almost certain if i can get Hyper-V to run initially then Docker will run.

Just out of interest what Security products are people using?? I use Sophos Central, the reason i ask this is i had 1 time where i had a VM start from a clean install and before i installed Sophos.  But ive not had time to test if Sophos had any effect.

Im going to try building a VM on Azure later using Win 10 ent 1709, and use this to run tests.  Ill post any findings.

Thanks

JK

**Comment by rvprasad** [comment]
Based on the error message, it seems the issue I am observing is a bit different from the one observed by @SCWells72.  So, I have created another issue #1224.

BTW, the behavior seems to be the same with the Docker release available on edge channel.

**Comment by rvprasad** [comment]
@jasonbivins I tried the command you suggested and I encountered the same message that you reported.

**Comment by Artgit** [comment]
I also faced several issues with virtualization after updating to Windows 10 1709 - https://stackoverflow.com/questions/46831762/maven-build-and-maven-failsafe-plugin-the-forked-vm-terminated-without-properl

https://superuser.com/questions/1260768/windows-10-update-to-1709-hyper-v-issue

**Comment by stritti** [comment]
Same problem on my machine.

**Comment by stritti** [comment]
After Update to Win 10 1709 I have same issue.

**Comment by SCWells72** [comment]
And now that I'm working from home again, the same problem occurs. I guess when I'm back in the office on Monday where it does work, I'll go ahead and clean out the virtual switch, etc., and let those reinstall/rebuild to see if that puts things back into a good state. I'll definitely keep folks posted here on my findings since it seems to be affecting a decent number of people at this point, albeit with multiple manifestations.

**Comment by PatrickLang** [comment]
@SCWells72 what kind of user are you logging in as? Microsoft account (email), local account, Azure Active Directory / work (still email), or Active Directory domain? 

Have you tried using a local admin user account to run Hyper-V manager instead?

cc @scooley

**Comment by MrMonkeyMonk** [comment]
No clue as to why @SCWells72 has differing behavior between home and work, but I was able to get linux containers working after removing the 'Windows Subsystem for Linux' feature.  After linux containers were working again, I was able to re-enable the Linux Subsystem.

**Comment by stritti** [comment]
Using Version: 17.10.0-ce (edge, Git commit: f4ffd25 Built: Tue Oct 17 19:09:12 2017) Docker works for me on Win10 (Update 1709) using the new Option to use "Linux Containers" (settings/general: last option flag).

**Comment by rvprasad** [comment]
@MrMonkeyMonk I tried your solution and it did not work.

**Comment by rvprasad** [comment]
@stritti I was able to reproduce your observation.  However, the failure persists with LCOW support disabled.

**Comment by stritti** [comment]
@rvprasad I think as already mentioned is a general problem of the Windows Update 1709: https://github.com/docker/for-win/issues/1221#issuecomment-338183298

**Comment by radbg** [comment]
Seems for me that when updating for some reason the MobyLinuxVM was removed from Hyper-V.

Unable to start: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: 'MobyLinuxVM' failed to start. (Virtual machine ID 753A72F4-C5E3-4423-8968-2269833DA6CB)

'MobyLinuxVM' failed to start worker process: %%3228369022 (0xC06D007E). (Virtual machine ID 753A72F4-C5E3-4423-8968-2269833DA6CB)
at Start-MobyLinuxVM, <No file>: line 291
at <ScriptBlock>, <No file>: line 386

**Comment by dopic** [comment]
I have the same problem with Docker, but I'm pretty sure that it's a problem with Hyper-V.
I can create the VM, but It doesn't start.

Also, I did some tests with an Ubuntu 17.10 VM, and it fails to start too, with the same problem.

My Hyper-V is running on Windows 10, update 1709, and I have Sophos installed. However removing Sophos, didn't solve the problem.

**Comment by Rakhmanov** [comment]
Had problem similar to @radbg 
Uninstalled Sophos and docker can start now.

**Comment by wdolek** [comment]
my 2¢: Had same issue and wasn't able to figure out how to make Docker work again. I ended up with complete reinstall of Windows 10 from recovery media from my computer vendor. Once Windows 10 got corrupted, "Reset" feature does not work (no matter if you try DISM).

... at least that worked for me.

(when I went trough this painful process, I enabled Hyper-V and containers **at the end** when 1709 has been installed (maybe that's also part of the trick)).

**Comment by surj08** [comment]
Patch tuesday on 11/14/17 fixed this (hyper v) issue right away

**Comment by rvprasad** [comment]
@surj08 If you are referring to build 1709 / 16299.64, then I just tried Docker 17.09.0-ce-win33 / stable  and it failed.  However, Docker 17.11.0-ce-rc3-win38 / edge works fine.  So, can you please share the versions of your Windows and Docker installations?

**Comment by surj08** [comment]
@rvprasad Yes Windows - 1709 / 16299.64  and Docker - 17.09.0-ce (not sure where the win came from but, I am using linux containers not windows)

Server:
 Version:      17.09.0-ce
 API version:  1.32 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:45:38 2017
 OS/Arch:      linux/amd64
 Experimental: true

I didn't realize that Sophos had been brought up in this thread but, I had previously uninstalled Sophos to rule it out but, now I am running the specified windows version and NO sophos. I will reinstall it now and report back

**Comment by TimHess** [comment]
@surj08 Were you able to get a response from Sophos? I seem to be having the same problem, also have Sophos installed (though its managed by IT and I can't remove it to test) with the same versions of Windows and Docker.

For me, switching to Docker Edge allows Docker to start up, but I wasn't able to actually run any images.

For what its worth, I'm on a brand new Dell Precision 5520 (just got it 2 days ago) and it came with Windows 1709

**Comment by surj08** [comment]
@TimHess I would not expect a quick response here. The other ticket I have open with them has already been a couple of weeks and after multiple escalations it seems work has stalled...

**Comment by atlastfree** [comment]
In my case the solution for "Docker for Windows" & "Hyper V" related problems after "Feature update to Windows 10, version 1709" update, was turning of the "Hyper V" feature  from within "Control Panel\All Control Panel Items\Programs and Features", restarting the computer and allowing Docker to enable the "Hyper V" feature again...

For reference the Docker error after 1709 update:
Unable to create: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Failed to create the virtual hard disk.

The system failed to create 'C:\Users\Public\Documents\Hyper-V\Virtual Hard Disks\MobyLinuxVM.vhdx'.
at New-MobyLinuxVM, <No file>: line 222
at <ScriptBlock>, <No file>: line 385
   at Docker.Backend.HyperV.RunScript(String action, Dictionary`2 parameters) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Backend\HyperV.cs:line 177
   at Docker.Backend.ContainerEngine.Linux.Start(Settings settings, String daemonOptions) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Backend\ContainerEngine\Linux.cs:line 123
   at Docker.Core.Pipe.NamedPipeServer.<>c__DisplayClass9_0.<Register>b__0(Object[] parameters) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Core\pipe\NamedPipeServer.cs:line 47
   at Docker.Core.Pipe.NamedPipeServer.RunAction(String action, Object[] parameters) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Core\pipe\NamedPipeServer.cs:line 145

72D12BC6-0E4B-4981-9907-A8F101E7159C/2017-11-18_15-55-10

**Comment by TimHess** [comment]
I'm on Win 10 Pro, can confirm that removing Sophos fixes the issue for me.

**Comment by mlewand** [comment]
I'm working on WIn 10 pro, it was upgraded from home to pro version just few months ago especially for Docker. After upgrade to version 1709 I have the same problem as OP. 😞 I'm not using any Sophos soft.

Couldn't google anything better than this thread, but no1 have figured out how to fix it though...

**Comment by jasonbivins** [comment]
@mlewand
 
in Hyper-V manager, can you create a VM (any vm at all) -- then start and _connect_ to it as well?

**Comment by mlewand** [comment]
@jasonbivins - not really, it seems that HyperV is unavailable. There are no vms displayed in Hyper-V manager.

I can see two errors being added to the event logger, each time I boot the os:

> The Host Compute Service failed to start:  General access denied error (0x80070005)

and

> Virtual Machine Management service failed to start.

I feel that the source of the issue is rather on Windows side rather than Docker.

**Comment by SCWells72** [comment]
@mlewand I'm seeing the same thing. The issue definitely seems to be with Windows/Hyper-V and not Docker. When I open Hyper-V on my work network (which was active when I first installed/configured Docker), all is fine. When I do the same thing on my home network, it's like Hyper-V isn't running on the local machine. And all this started with 1709. Everything worked perfectly before installing that update. The subsequent patches don't seem to have resolved that issue for me, though they have resolves some other issues that 1709 seemed to introduce.

**Comment by Deaton64** [comment]
Hi, I'm using Windows 10 Pro and usually use Virtual Box. Have a requirement to use Docker, but have the same issue as other people on here.
I use Sophos Endpoint for AV and Intercept X. I removed both of these and Hyper V works fine. Installed Eset and Intercept X and Hyper V doesn't work.  I'll try again with just Eset and if that works, I'll log a call with Sophos.

**Comment by kevinsheffield** [comment]
Having this issue as well, but removing Sophos is not an option for me.  Does anyone have an idea on another way around this?

**Comment by surj08** [comment]
I do have an update but, it's not a fix for this exactly... Sophos gave me the attached batch file. Run as admin and it uploads the file to Sophos Central for being added to the (Global) Exploit Mitigation Exclusion. I did this, added the exclusion, updated the agent, and it did not fix my base Hyper-V issue or docker. It did fix something for one of my clients, which was my main goal. However, that was a completely separate program. I am not sure what program/file path would have to be added to get this to work but, it's a starting point

[IXhyperV.txt](https://github.com/docker/for-win/files/1531294/IXhyperV.txt)

**Comment by kevinsheffield** [comment]
I found that script on a sophos forum post and it did not work for me either. Got to love the logic they have going in there though...

> rem add application under .wmv Shell\Open\Command key to make intercept x think that this application opens .wmv files

**Comment by surj08** [comment]
With the other app I had issues with, uninstalling Intercept X allowed it to work. For Hyper-V I'm having to remove the whole agent. 

Also, the application I got working with that script is no longer working despite the exclusion being in Central still

**Comment by mlewand** [comment]
**tl;dr** for me the source of issue was [control flow guard CFG](https://msdn.microsoft.com/en-us/library/windows/desktop/mt637065(v=vs.85).aspx).

I guess I have identified the source of the issue for me (as I mentioned I was not using anything like Sophos thingy).

With the release of Windows 10 ver. 1709 MS integrated mentioned CFG system. So far for me it was a source of pain, in particular it has slowed down some of my dev apps to a point where they were no longer usable. I have found a tutorial around the net with solution which was: disable the CFG - and in fact the performance issues went away. Now I figured out that this move was the source of the issue, here's how to restore CFG:

1. Open "Windows Defender Security Center".
![1_defendercenter](https://user-images.githubusercontent.com/5353898/33612295-651bd812-d9d1-11e7-9c1b-13d587cb4f42.png)
2. Go to "App & browser control".
![2_appcontrol](https://user-images.githubusercontent.com/5353898/33612304-6b9c6b34-d9d1-11e7-9e09-fea85cb0536c.png)
3. Scroll down to "Exploit protection" section, go to "Exploit protection settings".
![3_exploitprotectionsettings](https://user-images.githubusercontent.com/5353898/33612321-732d15a6-d9d1-11e7-9afa-f4c4be8d7a30.png)
4. Set "Control flow guard (CFG)" setting to On.
![4_enablecfg](https://user-images.githubusercontent.com/5353898/33612342-7f97578e-d9d1-11e7-9331-a30ea4d7778b.png)

After the change you might need to restart the PC. After that it was working for me just fine.

At the same time sorry to Docker team for offtoping this, but it's the first thread that has shown to me after googling this issue and had the most valuable information (it's much better than any MS related help tutorial, which should come at no surprise 😃).

**Comment by SCWells72** [comment]
@mlewand Bummer, I made that change and rebooted and still have the same issue. At this point I've completely uninstalled Docker, while on the network where it works removed the VM created by Docker so things are clean, uninstalled and reinstalled Hyper-V, etc., and it still doesn't work. Not sure what to try next.

**Comment by duffman03** [comment]
A team member is experiencing a similar looking error:
`Unable to create: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Generic failure
at New-Switch, <No file>: line 131
at <ScriptBlock>, <No file>: line 385
at Docker.Backend.HyperV.RunScript(String action, Dictionary`2 parameters) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Backend\HyperV.cs:line 177
at Docker.Backend.ContainerEngine.Linux.Start(Settings settings, String daemonOptions) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Backend\ContainerEngine\Linux.cs:line 123
at Docker.Core.Pipe.NamedPipeServer.<>c__DisplayClass9_0.<Register>b__0(Object[] parameters) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Core\pipe\NamedPipeServer.cs:line 47
at Docker.Core.Pipe.NamedPipeServer.RunAction(String action, Object[] parameters) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Core\pipe\NamedPipeServer.cs:line 145`

**One thing I've noted that I didn't see anyone mention here is that it seems no matter what we do the MobyLinuxVM file never gets generated.** We've tried multiple installation attempts of docker,  enabling and disabling of hyper-v(through windows features), changing permissions on the hyper-v directory(allowed all permissions to Everyone), confirmed virtualization is enabled(in task manager > performance), restarts at every step of the process, and running everything as administrator along the way.

Does this appear to be the same issue or should I file a separate ticket?

**Comment by Deaton64** [comment]
If you are using Intercept X, Sophos pointed me to: https://community.sophos.com/kb/en-us/127797

**Comment by Blottie** [comment]
I have been having the same problems with hyper-v since 1709.  Last night i did a fresh install of 1709 using the 1709 iso and all was working well, hyper v working great.  
This morning i joined my machine back to my domain and it has now got the same problem again, but on 1607 and 1703 being on the domain was not a problem for hyper-v.

**Comment by stojan-jovic** [comment]
mlewand's solution with changing setting for control flow guard (CFG) works form me. I put it off few days ago because of some problems with Git, but now needs to revert it and fixed problem with my Docker.

**Comment by Deaton64** [comment]
Sophos support have sorted my issue. They've run a batch file on my PC and that sends something back to Sophos Central so a rule can be added that ignores  c:\windows\system32\vmcompute.exe.
So I'm on Windows 1709 running Eset AV & Sophos Intercept X and the Sophos fix is the only thing I've changed on my PC and Docker runs as expected.
Not that it helps other people not running Sophos. My advice, if you're running Sophos, call Sophos support.

**Comment by stojan-jovic** [comment]
I'm not using Sophos... Anyway, I think that this is Windows related problem, everything other is just workaround.

**Comment by kevincam3** [comment]
same here, docker stopped working. Hyper-V is working because I have other things that use it, but Docker won't work anymore. I guess I'll have to go the VirtualBox route to see if that works.

**Comment by MagneticMule** [comment]
Curiously, at least in my case, this was an issue with a previously installed MobyLinuxVM. 

If you fire up Hyper-V Manager and look for MobyLinuxVM (Ver 8.0). On the actions panel, click the option to Remove Server. Fire up Docker again and it should start it's own MobyLinuxVM (Ver 8.2).

**Comment by kevincam3** [comment]
I found the problem was I had more than one Network Connection for the Hyper-V after the last Windows 10 upgrade. I just deleted all that were not used 

![image](https://user-images.githubusercontent.com/870934/34155564-fc09d66e-e4a0-11e7-8e57-c93d7128724b.png)

**Comment by LubosRemplik** [comment]
None of above worked. Had to use Edge channel instead of Stable channel then it works fine

**Comment by rogerfitz** [comment]
Reverted windows 1709 update fixed it for me

**Comment by djensen47** [comment]
@kevincam3 Did you delete "Default Switch"? That's a new Hyper-V feature that was added to the recent Insider build.

**Comment by kevincam3** [comment]
@djensen47,

Looking at my current Network Connections, it's still there. I have (Default Switch) and (DockerNAT)

**Comment by djensen47** [comment]
Do you recall which one you deleted? IIRC Default Switch cannot be deleted. But if you deleted DockerNAT, it may have been added by Docker when it found it missing.

**Comment by kevincam3** [comment]
I don't remember, I just know that I only left the two that I currently have and deleted all other Hyper-V entries.

**Comment by aalmada** [comment]
I can confirm that installing "edge" did fix the issue. I used the "network reset" before installation, just in case. After some hours of reading forums and failed attempts, I finally have Docker working on Windows 10 Enterprise on Boot Camp...

**Comment by Johannestegner** [comment]
Thought that a update with some information from Microsoft and the 17074 insider build:

> This build has an issue running Linux containers with Docker Community Edition for Windows (edge) on many PCs. The docker pull or docker run commands will fail with an error “Not enough memory resources are available to complete this operation”. We’re working on a fix for an upcoming Windows Insider build but you can work around the problem now with these steps:
> 1. Be sure “Docker for Windows” is running, then right-click the whale icon in the notification area and choose “settings”. 
> 2. On the General tab, uncheck “Enable Linux containers on Windows”. 
> 3. Next a dialog “Switch to Standard mode (Linux VM / Windows daemon” will pop up. Click OK. 
> 4. Wait for the whale icon to turn white.

In my case the following message appears when trying to run a linux container:
`ERROR: image operating system "linux" cannot be used on this platform`
But I guess it's a step in the right direction or something...

**Comment by AlexeiScherbakov** [comment]
I deleted all network devices via "netcfg -d", after it Docker started normal, but as expected I lost all wifi settings. After reboot docker is working normally.

**Comment by Ciantic** [comment]
I'm having a same issue, though I suspect this is Hyper-V related. If I open Hyper-V Manager application, try to connect to local it gives me error:

> An error occurred while attempting to connect to server "...". Check that the Virtual Machine Management service is running and that you are authorized to connect to the server.

I have so far in two days tried: 

* Checked, that Virtual Machine Management service is running
* Added myself to the Hyper-V admins group using `lusrmgr.msc`. 
* Re-installed Hyper-V (several times) feature from Windows settings, it didn't help.
* Tried to mess with DCOM rights
* Tried to enable DCOM Windows Firewall rules

Seems like there is only nuclear option left, re-installing Windows. Not looking forward to that.

**Comment by Ciantic** [comment]
Sorry for spamming, but god damit, that Control Flow Guard (CFG) thing worked! (The @mlewand's trick above)

I had disabled it when TortoiseGIT was having issues with it some months ago.

**Comment by hach-que** [comment]
Just to let everyone know, a workaround that definitely works (nothing else in this thread did), is to install [Docker Toolbox](https://docs.docker.com/toolbox/toolbox_install_windows/) instead of Docker for Windows.

**Comment by Hervian** [comment]
For me uninstalling Check point VPN solved the problem. It seems other has had the same problem, see [https://github.com/docker/for-win/issues/324#issuecomment-318911257](https://github.com/docker/for-win/issues/324#issuecomment-318911257)

**Comment by bunti2c** [comment]
If you are using Sophos, then try to disable the HitmanPro.Alert service and restart. This solved the problem for me.

**Comment by Asif-ak** [comment]
docker is behaving strangely after 1709. problem is when i tried to start docker without launching the hyper-v, it crashed. when i launched the hyper-v app, mobylinux vm was not there. When i start the docker with hyper-v app launched, docker started and somehow the mobylinux vm got back.

what on earth is going?

**Comment by nathanb12** [comment]
Currently using Sophos central, disabling hitmanPro.alert service and restarting fixed it for me.
Hopefully Sophos fix this, also had a similar issue with another app we use at work.

**Comment by Asif-ak** [comment]
I dont use Sophos

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=icon>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=link>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by TehJoE** [comment]
Disabling the HitmanPro.Alert service fixed it for me, but upon rebooting the system it re-enables itself. If you're quick you can stop the service before the VM boots up, but I didn't find any way to keep the service disabled. 

Evidently there is some kind of [workaround](https://community.sophos.com/products/intercept/early-access-preview/f/intercept-x-for-windows/97613/hyper-v-do-not-start-virtual-machines-windows-10-pro-1709/356481#356481) available that involves updating the HitmanPro.Alert service to a beta version. I personally don't know if it works, because I used a [different workaround](https://community.sophos.com/kb/en-us/118849).

**Comment by teoSeven** [comment]
Try unistall and install Hyper-v from windows features...works for me

**Comment by telkomops** [comment]
- Uninstalled the Docker For Windows
- Moved to Beta Channel. 
- Removed all folders in c:\users\username\.docker, 
- %appdata%\local\docker, 
- %appdata%\roaming\docker folders
- powershell in admin mode, ```Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Hyper-V -All```
- ```Enable-WindowsOptionalFeature -Online -FeatureName Containers -All```
- Restarted Hyper-V services.
- Installed Docker.
- Invoked Docker App with Run as Admin, when starting switched to Windows Containers and then switched back to Linux. 
Its a pain to get it running.

**Comment by igmwandama** [comment]
It might help someone, I experience the same issue and no solution could work.
Then I thought of adding my user account and that of local administrator to Hyper-V Administrator, after that docker is now working.

though not sure if its really a solution

**Comment by kingaj12** [comment]
Every time I restart Windows 10, the Docker service crashes.  It seems like all I can do is "reset to factory defaults", which removes all my Docker images.

Here is the report id:  1CD8E5DB-FA1C-4144-82D5-1A5F82520256/2018-03-13_08-48-02

**Comment by beawolf** [comment]
@telkomops, your solution worked for me. Thank you.

**Comment by pldmgg** [comment]
(I also left this comment on #606 ... I hope that's okay)

Just wanted to share a workaround that's worked for me 100% of the time thus far (knock-on-wood) when starting Docker For Windows after baremetal host reboot. Note that this has only been tested with docker-for-windows 17.12.0.15139 using Hyper-V.

I wrote several PowerShell functions that you can load by running the below in an Elevated PowerShell prompt (please feel free to review the code in the gist if you feel at all uncomfortable...it's all pretty self-explanatory).

IMPORTANT NOTE: Before running the `Start-DockerAfterReboot` function below, make sure you uncheck the "Start Docker at login" checkbox in the "Docker For Windows.exe" GUI. (I'd love it if there were a way to do this from the cmdline...)

IMPORTANT NOTE: The other manual step you need to do before running this is making sure that all Hyper-V VMs created via docker-machine are 'Off'.

    $GistUrl = "https://gist.githubusercontent.com/pldmgg/92b7a6975a7765293579a9bec296aec1/raw/6275e33d717c2623d548537c6e11165fc17045b1/DockerForWindowsManagement.ps1"
    $OutFileLocation = "$HOME\Downloads\DockerForWindowsManagement.ps1"
    Invoke-WebRequest -Uri $GistUrl -OutFile $OutFileLocation
    . $OutFileLocation
    Start-DockerAfterReboot

Basically, the `Start-DockerAfterReboot` function does the following:

- Prompts the user to confirm that they performed the two manual steps I mentioned above
- Stops "Docker For Windows.exe" if it is running
- Stops the com.docker.service if it is running
- Changes the com.docker.service StartupType from 'Automatic' to 'Manual' if it's not already
- Sleeps for about 2 minutes and prompts the user to confirm that the Docker For Windows GUI says that Docker is up and running (again...I wish that there was a way to check for this from the cmdline so that you don't have to respond to the prompt).
- Starts/Restarts all docker-machine VMs (in parallel) in a loop until they actually get IPs (I don't know about you guys, but this was necessary because my docker-machine VMs don't consistently get IPs after restart...but they will eventually after a few reboots...)

Hope this can help! And I really hope these issues can be addressed soon...I feel like they've been around for a little too long and they're not necessarily that difficult to fix...

(P.S. Shout out to @RamblingCookieMonster for the `Invoke-Parallel` PowerShell function)

**Comment by redBirdTx** [comment]
I too had similar issue on windows 10 after applying a update. The problem resolves after uninstall docker and reinstall

**Comment by Johannestegner** [comment]
Update: 18.03.0 runs fine on win 1803 (17133.1). 
Guessing that the following bugfix stated in changelog is the "fix" at the least for my issue.

> Fix Linuxkit start on Windows RS4 Insider. Fixes docker/for-win#1458, docker/for-win#1514, docker/for-win#1640

**Comment by 99linesofcode** [comment]
Have been experiencing the same issue on Win 10 Pro with Hyper-V and Docker for Windows CE (both stable and edge channels).

In my case, Hyper-V was unable to connect to the MobyLinuxVM server and the only thing that really helped was to completely disabling and enabling Hyper-V through `Windows Features`. Curiously, the PowerShell commands didn't resolve the issue for me.

Though I suspect that this is an issue with Hyper-V, it could be that some people replying here are experiencing issues caused by duplicated network adapters or a corrupt MobyLinuxVM. I'm including this SO question as it underlines some of the replies given in this thread.

Hopefully someone with a better understanding of the Docker internals is able to resolve this issue as I was still experiencing this on 1804 edge under Windows 1709 (which is the latest update according to my dev box)

✌️

**Comment by bogmoroz** [comment]
@telkomops thank you!

**Comment by fquinner** [comment]
Well, that was fun.

Looks like there are several distinct issues... for me, I was getting something along the lines of:

`Hyper-V\Get-VMHost : Hyper-V encountered an error trying to access an object on computer 'localhost' because the
object was not found. The object might have been deleted. Verify that the Virtual Machine Management service on the
computer is running.`

It turns out that at some point in time, windows has somehow changed the host name, but not the netbios / logonserver name so even though the ps1 scripts referred to `localhost`, it wouldn't resolve correctly to the hyper-v service.

You can tell that this is your problem if you drop to the command prompt and any of these values print the pre-modification host name:

```
> hostname
> echo %COMPUTERNAME%
> echo %LOGONSERVER%
> echo %USERDOMAIN%
> echo %USERDOMAIN_ROAMINGPROFILE%
```

If they're all the same, this is probably not the problem you're facing and this solution probably won't work for you.

I did hack the ps1 scripts to compensate which worked OK, but then figured the less invasive option would be to uninstall, fix the hostname consistency in windows, then reinstall which seemed to work.

You can fix the hostname by simply changing the host to something else then changing it back.

**Comment by gareth-horstmann** [comment]
@fquinner thank you, that was my exact problem. My hostname was different and changing my computer name to match the other variables resolved it.

**Comment by kshittiz** [comment]
I was able to fix this issue in 3 steps:
1. disable hyper-v & containers from "turn windows features on/off"
2. restart computer & start windows power shell as administrator
3. execute this command:  Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Hyper-V -All

This command will enable hyper-v. No need to enable containers.
hope this helps!

**Comment by suhas1git** [comment]
You can try uninstalling ,clear all the data left out in the Box then install agian and try .it worked for me

**Comment by coatsnmore** [comment]
`netcfg -d` cleaned it up for me and docker starts normally.

**Comment by LiyueWang** [comment]
upgraded Windows to 1803, and the problem solved.

**Comment by MattWalsh99** [comment]
I have a similar issue on Server 2K16 (WindowsServerCore:1803)

Installed Hyper-V in the container. OS reports as 'Installed', pending reboot.  Restarted (Stopped/Started container) hyper-v is no longer reported as installed, just available.

I have even recommitted to a new image and started a new container, same problems.

**Comment by nigamelastic** [comment]
Solved the same issue by simply checking the experimental features(mine was unchecked by default) which restarted my docker. I ran the command again (a fresh batch) and it worked :)

**Comment by shaulbehr** [comment]
A warning to those who might think of uninstalling and reinstalling Hyper-V: I just did that, and it half-worked. The uninstall worked; the reinstall doesn't. No matter how I try reinstall Hyper-V, whether with UI or elevated PowerShell command, Windows reboots and says "Couldn't install updates", then backs out my changes, and now I'm stuck without Docker and without Hyper-V either. 
<feels sorry for self/>

**Comment by AbsSimon** [comment]
had the same problem, and problem solved after

1.     disable hyper-v & containers from "turn windows features on/off"
2.     restart computer & start windows power shell as administrator
3.     execute this command: Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Hyper-V 

thank you kshittiz ;)

**Comment by Funkrusha** [comment]
In Short:
I lost my hypervisor boot command during disk upgrade.
The Solution:
Reenable hypervisor Launch on boot command
`bcdedit /set {current} hypervisorlaunchtype auto`

More Details:
I got problems after ugrading the internal nvme disk and switched to a bigger one.
Since booting with the new disk docker was unable to start. The docker message told me that i Need to enable hyper-v and protected data execution. As i tried to run normal vm‘s in Hyper-V, i realized that Hyper-V VM‘s can‘t run. As i checken my boot command i noticed that i lost the entry for hypervisorlaunchtype.

**Comment by davidmadi** [comment]
Disable the "Containers" installation in windows.
Restart
------
Or
Uninstall everything about docker.
Uninstall hyper-v (windows)
Uninstall containers (windows)
Restart
Install docker
Restart
Install Hyper-V (full)
Restart
Finish

**Comment by matham** [comment]
I had the same error so I looked under Event Viewer and saw a error in Applications/Windows/Hyper-V Compute Service saying `Windows could not start the Hyper-V Host Compute Service on Local Computer. Error 0x80070005: Access is denied`. 

That error brought me to https://serverfault.com/questions/886357/error-0x80070005-access-denied-when-first-start-hyper-v-in-windows-10-pro/887190.

On that page it says to change `Control flow guard (CFG)` to be `On by default`, which I did and now docker works fine!!!

**Comment by dusnoki** [comment]
> I had the same error so I looked under Event Viewer and saw a error in Applications/Windows/Hyper-V Compute Service saying `Windows could not start the Hyper-V Host Compute Service on Local Computer. Error 0x80070005: Access is denied`.
> 
> That error brought me to https://serverfault.com/questions/886357/error-0x80070005-access-denied-when-first-start-hyper-v-in-windows-10-pro/887190.
> 
> On that page it says to change `Control flow guard (CFG)` to be `On by default`, which I did and now docker works fine!!!

Yup... this did it for me :)

**Comment by shaulbehr** [comment]
FTR, my problem turned out to have been caused by some internet filtering software I had installed. When I disabled the filter program, Docker ran fine.

**Comment by yoavbls** [comment]
@mlewand
You are my hero 😍

**Comment by edisondotme** [comment]
Following the instructions [here](https://shauncassells.wordpress.com/2018/05/24/fix-hyper-v-encountered-an-error-trying-to-access-an-object-on-computer-localhost-because-the-object-was-not-found-the-object-might-have-been-deleted-verify-that-the-virtual-machine-managemen/) worked perfectly for me.

I just had to run this command in an elevated powershell prompt:
`MOFCOMP %SYSTEMROOT%\System32\WindowsVirtualization.V2.mof`

**Comment by Karlheinzniebuhr** [comment]
I had to run `bcdedit /set hypervisorlaunchtype off` without rebooting, I couldn't start Docker for Windows (error: cannot open vmms service on computer. If you just enabled Hyper-V, please restart now.) but managed to finally run docker from the command line!

**Comment by Puiu** [comment]
It worked for me to switch to Windows Containers and then back to Linux Containers. It was a little misleading since I got the same error message as when virtualization on the VM was disabled.
I am running it on a VM with Windows 2016 DataCenter edition.

**Comment by os72** [comment]
> 
> 
> Following the instructions [here](https://shauncassells.wordpress.com/2018/05/24/fix-hyper-v-encountered-an-error-trying-to-access-an-object-on-computer-localhost-because-the-object-was-not-found-the-object-might-have-been-deleted-verify-that-the-virtual-machine-managemen/) worked perfectly for me.
> 
> I just had to run this command in an elevated powershell prompt:
> `MOFCOMP %SYSTEMROOT%\System32\WindowsVirtualization.V2.mof`

This was it, thanks!

**Comment by jfmmm** [comment]
> > hostname
> > echo %COMPUTERNAME%
> > echo %LOGONSERVER%
> > echo %USERDOMAIN%
> > echo %USERDOMAIN_ROAMINGPROFILE%

This gave me:

MSI
MSI
\\\\MSI
MSI
MSI

Does that count as different?

**Comment by METACEO** [comment]
Thank you @edisondotme ! I had uninstalled and re-installed docker and was still encountering this error until running your line!

**Comment by piMinusE** [comment]
Thank you for your suggestion to check hostname etc, @fquinner !
(https://github.com/docker/for-win/issues/1221#issuecomment-381915342)

In my case. the Windows hostname was longer than 15 characters, so the output of 
c:> hostname
didn't match %COMPUTERNAME%

The solution was to shorten the hostname and reboot, after which Docker started working again. 

Note that this seems to be due to a recent (early April 2019) Windows 10 update.  Docker was working until recently, but the recent update killed both Docker and the Hyper-v manager.

**Comment by knsakib** [comment]
Emil's solution from here https://stackoverflow.com/questions/48255871/failed-to-start-the-virtual-machine-mobylinuxvm-because-one-of-the-hyper-v-com resolved my issue. 

"Here is a solution if you are getting this error on Azure Windows 10 VM where you have installed Docker:

Ensure Windows Hyper-V featutes are enabled by running PowerShell cmdlet:
Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Hyper-V -All -Verbose

Ensure Windows Containers feature is enabled by running PowerShell cmdlet:
Enable-WindowsOptionalFeature -Online -FeatureName Containers -All -Verbose

Ensure Hypervisor is set to auto start in the Boot Configuration Database (BCD) by running in elevated command prompt the command:
bcdedit /set hypervisorlaunchtype Auto

After running all of the above and you restart the Azure VM, Docker should be starting normally.

In addition to the Answer above the PowerShell cmdlet to set the Virtual Machine Setting is:

Set-VMProcessor -VMName <Enter-VM-Name> -ExposeVirtualizationExtensions $true -Verbose

**Comment by rvvincelli** [comment]
Hi all,

Just applied the following from our IT desk and it seems to work - Docker starts normally:

_To use Hyper-V on your Windows 10 client follow the steps below:
•         Install Hyper-V via Programs and Features
o   Reboot
•         Open “App & Browser Control”
•         Scroll down to “Exploit Protection” and click “Exploit protection settings”
•         Change “Control flow guard (CFG)” to “Off by default”
o   Reboot
•         Open “App & Browser Control”
•         Scroll down to “Exploit Protection” and click “Exploit protection settings”
•         Change “Control flow guard (CFG)” to “On by default”
o   Reboot
•         Start Hyper-V
o   Configure virtual switch
o   Create VM (Gen2)
•         Start vmconnect.exe as administrator (Pin it to taskbar for quick access)
o   Connect to localhost and select the created virtual machine
Thanks,_

Thank you!

**Comment by Deaton64** [comment]
It doesn’t help any issues in this thread, but is what I’ve been looking forward to.
Windows Linux Subsystem 2 will support Docker containers.

https://www.neowin.net/news/microsoft-announces-windows-terminal-and-a-major-update-to-windows-subsystem-for-linux

**Comment by AskanSimon** [comment]
a little bit offtopic, i had also minor problems with varnish and memory consumption, and made a dual installation of ubuntu 18.04 lts (dont use 19.04, to early ...) and then you feel, docker is made for linux (but docker for windows is way better than docker for mac). so there is a plan b, if nothing has helped.

**Comment by michelefalconi** [comment]
I resolved it following this link:
[https://social.technet.microsoft.com/Forums/en-US/ee5b1d6b-09e2-49f3-a52c-820aafc316f9/hyperv-doesnt-work-after-upgrade-to-windows-10-1809?forum=win10itprovirt](https://social.technet.microsoft.com/Forums/en-US/ee5b1d6b-09e2-49f3-a52c-820aafc316f9/hyperv-doesnt-work-after-upgrade-to-windows-10-1809?forum=win10itprovirt)

Summary:
1, Open "Window Security"
2, Open "App & Browser control"
3, Click "Exploit protection settings" at the bottom
4, Switch to "Program settings" tab
5, Locate "C:\WINDOWS\System32\vmcompute.exe" in the list and expand it
6, Click "Edit"
7, Scroll down to "Code flow guard (CFG)" and uncheck "Override system settings"
8, Start vmcompute from powershell "net start vmcompute"

Michele

**Comment by joesixpack** [comment]
None of the above fixes worked.  However, this did:

net localgroup administrators LOCAL SERVICE /add
net localgroup administrators LOCAL /delete
net localgroup administrators SERVICE /delete

Then reboot.

**Comment by zekth** [comment]
Just fixed this case on my env, i had the peripheral "enumerator of microsoft virtual drive" disabled (litteral translation).

**Comment by foggy-glasses** [comment]
> 
> 
> I resolved it following this link:
> https://social.technet.microsoft.com/Forums/en-US/ee5b1d6b-09e2-49f3-a52c-820aafc316f9/hyperv-doesnt-work-after-upgrade-to-windows-10-1809?forum=win10itprovirt
> 
> Summary:
> 1, Open "Window Security"
> 2, Open "App & Browser control"
> 3, Click "Exploit protection settings" at the bottom
> 4, Switch to "Program settings" tab
> 5, Locate "C:\WINDOWS\System32\vmcompute.exe" in the list and expand it
> 6, Click "Edit"
> 7, Scroll down to "Code flow guard (CFG)" and uncheck "Override system settings"
> 8, Start vmcompute from powershell "net start vmcompute"
> 
> Michele

Thanks! This worked for me and I tried quite a few options before this worked. One of the previous suggestions only mentioned about changing the Exploit Protection Settings.

**Comment by oliverfei** [comment]
Enabling Windows containers feature worked for me.

**Comment by cosloli** [comment]
Me too.

**Comment by joesixpack** [comment]
Looks like your Hyper-V isn't installed/running.

**Comment by bhupendra23** [comment]
> Looks like your Hyper-V isn't installed/running.

its runing.

**Comment by sirudog** [comment]
This solved the issue for me:
https://github.com/docker/for-win/issues/2090#issuecomment-567590149

**Comment by HardToGiveaName** [comment]
> > I resolved it following this link:
> > https://social.technet.microsoft.com/Forums/en-US/ee5b1d6b-09e2-49f3-a52c-820aafc316f9/hyperv-doesnt-work-after-upgrade-to-windows-10-1809?forum=win10itprovirt
> > Summary:
> > 1, Open "Window Security"
> > 2, Open "App & Browser control"
> > 3, Click "Exploit protection settings" at the bottom
> > 4, Switch to "Program settings" tab
> > 5, Locate "C:\WINDOWS\System32\vmcompute.exe" in the list and expand it
> > 6, Click "Edit"
> > 7, Scroll down to "Code flow guard (CFG)" and uncheck "Override system settings"
> > 8, Start vmcompute from powershell "net start vmcompute"
> > Michele
> 
> Thanks! This worked for me and I tried quite a few options before this worked. One of the previous suggestions only mentioned about changing the Exploit Protection Settings.

It does work! Thranks

**Comment by dariusgm** [comment]
> I resolved it following this link:
> https://social.technet.microsoft.com/Forums/en-US/ee5b1d6b-09e2-49f3-a52c-820aafc316f9/hyperv-doesnt-work-after-upgrade-to-windows-10-1809?forum=win10itprovirt
> Summary:
> 1, Open "Window Security"
> 2, Open "App & Browser control"
> 3, Click "Exploit protection settings" at the bottom
> 4, Switch to "Program settings" tab
> 5, Locate "C:\WINDOWS\System32\vmcompute.exe" in the list and expand it
> 6, Click "Edit"
> 7, Scroll down to "Code flow guard (CFG)" and uncheck "Override system settings"
> 8, Start vmcompute from powershell "net start vmcompute"
> Michele
> 

Thanks, also had to do this on a fresh installed Windows 10 pro build 1909
